packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -68,60 +70,170 @@ export enum Protocol {
 }
 
 /**
- * Properties used to define healthchecks when creating virtual nodes.
- * All values have a default if only specified as {} when creating.
- * If property not set, then no healthchecks will be defined.
+ * Properties used to define healthchecks.
  */
-export interface HealthCheck {
+interface HealthCheckCommonOptions {
   /**
-   * Number of successful attempts before considering the node UP
+   * The number of consecutive successful health checks that must occur before declaring listener healthy.
    *
    * @default 2
    */
   readonly healthyThreshold?: number;
 
   /**
-   * Interval in milliseconds to re-check
+   * The time period between each health check execution.
    *
-   * @default 5 seconds
+   * @default Duration.seconds(5)
    */
   readonly interval?: cdk.Duration;
 
   /**
-   * The path where the application expects any health-checks, this can also be the application path.
+   * The amount of time to wait when receiving a response from the health check.
    *
-   * @default /
+   * @default Duration.seconds(2)
    */
-  readonly path?: string;
+  readonly timeout?: cdk.Duration;
 
   /**
-   * The TCP port number for the healthcheck
+   * The number of consecutive failed health checks that must occur before declaring a listener unhealthy.
    *
-   * @default - same as corresponding port mapping
+   * @default - 2
    */
-  readonly port?: number;
+  readonly unhealthyThreshold?: number;
+}
 
+/**
+ * Properties used to define HTTP Based healthchecks.
+ */
+export interface HttpHealthCheckOptions extends HealthCheckCommonOptions {
   /**
-   * The protocol to use for the healthcheck, for convinience a const enum has been defined.
-   * Protocol.HTTP or Protocol.TCP
+   * The destination path for the health check request.
    *
-   * @default - same as corresponding port mapping
+   * @default /
    */
-  readonly protocol?: Protocol;
+  readonly path?: string;
+}
+
+/**
```

Comment:
Can we move all of the newly created classes and interfaces into their own, separate file?